src/parser.rs

```diff
@@ -10,6 +11,8 @@ use reader::{Reader, ReaderOffset};
 /// An error that occurred when parsing.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum Error {
+    /// An I/O error occurred while reading.
+    Io,
```

Comment:
Would it be super difficult to remove the `derive(Copy)` so we could embed the `io::Error` inside here? Just a thought, not a blocker for anything.